[receiver/mongodb] Panic when version is not determined

### Component(s)

receiver/mongodb

### What happened?

See https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5419415606/jobs/9852499795?pr=23858

```
2023/06/30 05:04:34 🚧 Waiting for container id 96f9ec7bcb34 image: 8b1c8ba1-5072-4331-bd72-a77b19c857c7:56f9298e-d004-4bbc-9950-9e16f3fe1ed2. Waiting for: &{Port:27017 timeout:0xc000483858 PollInterval:100ms}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6cd536]

goroutine 514 [running]:
github.com/hashicorp/go-version.(*Version).String(0x0)
	/home/runner/go/pkg/mod/github.com/hashicorp/go-version@v1.6.0/version.go:369 +0x96
github.com/hashicorp/go-version.(*Version).Compare(0x0, 0xc0004a9c70)
	/home/runner/go/pkg/mod/github.com/hashicorp/go-version@v1.6.0/version.go:116 +0x52
github.com/hashicorp/go-version.(*Version).LessThan(...)
	/home/runner/go/pkg/mod/github.com/hashicorp/go-version@v1.6.0/version.go:308
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver.(*mongodbScraper).recordLockAcquireCounts(0xc00038c660, 0xc00028de8b?, 0x5?, {0xc00028de8b, 0x5}, 0xc0001adb60)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/metrics.go:356 +0xca
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver.(*mongodbScraper).recordNormalServerStats(0xc0001aeb60?, 0x1b1ab08?, 0xc0001fbaa0?, {0xc00028de8b, 0x5}, 0x0?)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/scraper.go:167 +0xdd
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver.(*mongodbScraper).collectDatabase(0xc00038c660, {0x1b1ab08, 0xc0001fbaa0}, 0x4?, {0xc00028de8b, 0x5}, 0xc0001adb60)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/scraper.go:115 +0x58e
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver.(*mongodbScraper).collectMetrics(0xc00038c660, {0x1b1ab08, 0xc0001fbaa0}, 0xc0001adb60)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/scraper.go:89 +0x512
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver.(*mongodbScraper).scrape(0xc00038c660, {0x1b1ab08, 0xc0001fbaa0})
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/scraper.go:71 +0x2f9
go.opentelemetry.io/collector/receiver/scraperhelper.ScrapeFunc.Scrape(...)
	/home/runner/go/pkg/mod/go.opentelemetry.io/collector/receiver@v0.80.1-0.20230629144634-c3f70bd1f8ea/scraperhelper/scraper.go:20
go.opentelemetry.io/collector/receiver/scraperhelper.(*controller).scrapeMetricsAndReport(0xc000003680, {0x1b1aa98, 0xc000122008})
	/home/runner/go/pkg/mod/go.opentelemetry.io/collector/receiver@v0.80.1-0.20230629144634-c3f70bd1f8ea/scraperhelper/scrapercontroller.go:210 +0x34e
go.opentelemetry.io/collector/receiver/scraperhelper.(*controller).startScraping.func1()
	/home/runner/go/pkg/mod/go.opentelemetry.io/collector/receiver@v0.80.1-0.20230629144634-c3f70bd1f8ea/scraperhelper/scrapercontroller.go:192 +0x228
created by go.opentelemetry.io/collector/receiver/scraperhelper.(*controller).startScraping
	/home/runner/go/pkg/mod/go.opentelemetry.io/collector/receiver@v0.80.1-0.20230629144634-c3f70bd1f8ea/scraperhelper/scrapercontroller.go:173 +0x8f
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver	55.[266](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5419415606/jobs/9852499795?pr=23858#step:5:267)s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver/internal/metadata	0.031s [no tests to run]
FAIL
make[2]: *** [../../Makefile.Common:115: mod-integration-test] Error 1
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver'
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
make[1]: *** [Makefile:176: receiver/mongodbreceiver] Error 2
make: *** [Makefile:78: integration-test] Error 2
Error: Process completed with exit code 2.
```

### Collector version

0.80.0

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/.chloggen/fix-mongodb-version-missing.yaml
+++ b/.chloggen/fix-mongodb-version-missing.yaml
@@ -1,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: mongodbreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix missing version in mongodb creating panics during scrapes
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [23859]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/receiver/mongodbreceiver/scraper.go
+++ b/receiver/mongodbreceiver/scraper.go
@@ -30,10 +30,12 @@ type mongodbScraper struct {
 }
 
 func newMongodbScraper(settings receiver.CreateSettings, config *Config) *mongodbScraper {
+	v, _ := version.NewVersion("0.0")
 	return &mongodbScraper{
-		logger: settings.Logger,
-		config: config,
-		mb:     metadata.NewMetricsBuilder(config.MetricsBuilderConfig, settings),
+		logger:       settings.Logger,
+		config:       config,
+		mb:           metadata.NewMetricsBuilder(config.MetricsBuilderConfig, settings),
+		mongoVersion: v,
 	}
 }
 

